src/server/MapleStatEffect.java

```diff
@@ -1088,6 +1088,8 @@ private void applyBuffEffect(MapleCharacter applyfrom, MapleCharacter applyto, b
             } else if (isDs()) {
                 List<Pair<MapleBuffStat, Integer>> dsstat = Collections.singletonList(new Pair<>(MapleBuffStat.DARKSIGHT, 0));
                 mbuff = MaplePacketCreator.giveForeignBuff(applyto.getId(), dsstat);
+            } else if (isWKCharge()){
```

Comment:
I would advise to look over why these skills are guarded by the other if block. If it is possible to re-write some of this code.

Reply:
As far as I can tell, these are cases where the packet needs to be modified to match the skill that the MapleStatEffect object corresponds to. Aside from us mucking with already-defined skill constants, there is no realistically possible way for any skill to match more than one of the cases listed there. I see no reason to rewrite this class for now.